Entities/AtomGroup.cpp

```diff
@@ -1,2618 +1,1750 @@
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void AtomGroup::Clear() {
+		m_Atoms.clear();
+		m_SubGroups.clear();
+		m_OwnerMO = nullptr;
+		m_Material = g_SceneMan.GetMaterialFromID(g_MaterialAir);
+		m_AutoGenerate = false;
+		m_Resolution = 1;
+		m_Depth = 0;
+		m_JointOffset.Reset();
+		m_LimbPos.Reset();
+		m_MomentOfInertia = 0;
+		m_IgnoreMOIDs.clear();
 	}
-    else if (m_pMaterial->GetIndex() != m_Atoms.front()->GetMaterial()->GetIndex())
-        m_pMaterial = m_Atoms.front()->GetMaterial();
-
-    return 0;
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          Create
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Creates a AtomGroup to be identical to another, by deep copy.
-
-int AtomGroup::Create(const AtomGroup &reference)
-{
-    return Create(reference, false);
-}
-
-int AtomGroup::Create(const AtomGroup &reference, bool onlyCopyOwnerAtoms)
-{
-    Entity::Create(reference);
-
-    m_AutoGenerate = false; //Don't autogenerate because we'll copy the atoms below
-    m_pMaterial = reference.m_pMaterial;
-    m_Resolution = reference.m_Resolution;
-    m_Depth = reference.m_Depth;
-
-	m_SubGroups.clear();
-	for (const Atom * atom : reference.m_Atoms)
-	{
-		if (!onlyCopyOwnerAtoms || atom->GetSubID() == 0)
-		{
-			Atom *pAtomCopy = new Atom(*atom);
-			pAtomCopy->SetIgnoreMOIDsByGroup(&m_IgnoreMOIDs);
-
-			m_Atoms.push_back(pAtomCopy);
-
-			// Add to the appropriate spot in the subgroup map
-			long int subID = pAtomCopy->GetSubID();
-			if (subID != 0)
-			{
-				// Make a new list for the subgroup ID if there isn't one already
-				if (m_SubGroups.find(subID) == m_SubGroups.end())
-				{
-					m_SubGroups.insert(pair<long int, list<Atom *>>(subID, list<Atom *>()));
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int AtomGroup::Create() {
+		if (Entity::Create() < 0) {
+			return -1;
+		}
+		if (!m_AutoGenerate) {
+			// Needs to be set manually by the new MO owner.
+			m_OwnerMO = nullptr;
+			m_Resolution = 0;
+		}
+		// Make sure we have at least one Atom in the group, and that we have a proper material assigned
+		if (m_Atoms.empty()) {
+			Atom *atom = new Atom(Vector(), m_Material, m_OwnerMO);
+			m_Atoms.push_back(atom);
+		} else if (m_Material->GetIndex() != m_Atoms.front()->GetMaterial()->GetIndex()) {
+			m_Material = m_Atoms.front()->GetMaterial();
+		}
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int AtomGroup::Create(const AtomGroup &reference, bool onlyCopyOwnerAtoms) {
+		Entity::Create(reference);
+
+		m_OwnerMO = nullptr; // Needs to be set manually by the new MO owner.
```

Comment:
I guess we don't if we construct a new AtomGroup before calling this (since we don't use the copy constructor anywhere) but we do if we repurpose an existing one.

Reply:
Oh true, I guess we could technically clone create without clearing an old one. Super edge case but no harm in it I guess.